datafusion/sql/src/utils.rs

```diff
@@ -287,6 +287,28 @@ pub(crate) fn value_to_string(value: &Value) -> Option<String> {
     }
 }
 
+pub(crate) fn transform_bottom_unnests(
+    input: &LogicalPlan,
+    unnest_placeholder_columns: &mut Vec<String>,
+    inner_projection_exprs: &mut Vec<Expr>,
+    original_exprs: &[Expr],
+) -> Result<Vec<Expr>> {
+    Ok(original_exprs
```

Comment:
This does the recursion manually -- I wonder if it would be possible / useful to do the recursion using `transform_up` from TreeNode: https://docs.rs/datafusion/latest/datafusion/common/tree_node/trait.TreeNode.html#method.transform_up

Reply:
Thank you @alamb for your review. I'm not quite sure what you mean, maybe you can add more description, thanks again. 🙏

Reply:
I think I was confused -- sorry

Reply:
I think we can, in fact transform_up_down is needed, i'll introduce why in this PR when it's ready: https://github.com/apache/datafusion/pull/11577